lib/puppet/provider/ec2_securitygroup/v2.rb

```diff
@@ -127,12 +144,14 @@ def create
     ) if resource[:tags]
 
     @property_hash[:id] = response.group_id
-    rules = resource[:ingress]
-    authorize_ingress(rules)
+    ingress_rules = resource[:ingress]
+    egress_rules = resource[:egress]
+    authorize_rules(ingress_rules, 'ingress')
+    authorize_rules(egress_rules, 'egress')
```

Comment:
Using symbols (`:egress`) instead of strings is a useful microoptimisation, since this will index into rubies internal symbol cache instead of creating a new string. Symbols are also immutable, making everything safer.